cmake/CPM.cmake

```
@@ -42,7 +42,11 @@ if(NOT COMMAND cpm_message)
   endfunction()
 endif()
 
-set(CURRENT_CPM_VERSION 1.0.0-development-version)
+if("${PROJECT_NAME}" STREQUAL "CPM" AND CPM_RELEASE)
+  set(CURRENT_CPM_VERSION "${PROJECT_VERSION}")
+else()
+  set(CURRENT_CPM_VERSION 1.0.0-development-version)
+endif()
```

```suggestion
if(DEFINED CPM_VERSION)
  set(CURRENT_CPM_VERSION "${CPM_VERSION}${CPM_DEVELOPMENT}")
else()
  set(CURRENT_CPM_VERSION 1.0.0-development-version)
endif()
```
This should be eliminated altogether. Maybe even move away from `get_cpm` if this approach is working seamlessly.

---

Yes get_cpm could be suppressed in future I think but it's nice to have time to make the move

---

${CPM_DEVELOPMENT} is for future developement ? when workflow will be changed?